stan/math/prim/mat/fun/ocl_gpu.hpp

```diff
@@ -0,0 +1,296 @@
+#ifndef STAN_MATH_PRIM_MAT_FUN_OCL_HPP
```

Comment:
I was not really sure where to put this file. It's purpose is to set up the gpu backend. Should this be in some more internal folder?

Reply:
let me think about it. We have the autodiff stack sitting in `stan/math/memory/stack_alloc.hpp`.